sky/jobs/utils.py

```diff
@@ -750,20 +792,28 @@ def cancel_job_by_name(cls, job_name: str) -> str:
         return cls._build(code)
 
     @classmethod
-    def stream_logs_by_name(cls, job_name: str, follow: bool = True) -> str:
-        code = textwrap.dedent(f"""\
-        msg = utils.stream_logs_by_name({job_name!r}, follow={follow})
-        print(msg, flush=True)
+    def stream_logs(cls,
+                    job_name: Optional[str],
+                    job_id: Optional[int],
+                    follow: bool = True,
+                    controller: bool = False) -> str:
+        # We inspect the source code of the function here for backward
+        # compatibility.
+        # TODO: change to utils.stream_logs(job_id, job_name, follow) in v0.8.0.
+        # Import libraries required by `stream_logs`
+        code = textwrap.dedent("""\
+        import os
+
+        from sky.skylet import job_lib, log_lib
+        from sky.skylet import constants
+        from sky.jobs.utils import stream_logs_by_id
+        from typing import Optional
         """)
-        return cls._build(code)
+        code += inspect.getsource(stream_logs)
+        code += textwrap.dedent(f"""\
 
-    @classmethod
-    def stream_logs_by_id(cls,
-                          job_id: Optional[int],
-                          follow: bool = True) -> str:
-        code = textwrap.dedent(f"""\
-        job_id = {job_id} if {job_id} is not None else state.get_latest_job_id()
-        msg = utils.stream_logs_by_id(job_id, follow={follow})
+        msg = stream_logs({job_id!r}, {job_name!r}, 
```

Comment:
```suggestion
        msg = utils.stream_logs({job_id!r}, {job_name!r}, 
```
should this be `utils.stream_logs`? I dont saw an `from sky.jobs.utils import stream_logs` here